### PR TITLE
EL-1102: Make migration rake task more robust

### PR DESCRIPTION
**Pull request by patrick-laa**
[Jira ticket](https://dsdmoj.atlassian.net/browse/EL-1102)

## What changed and why

Allow migrations to recover from an error that sometimes happens when 2 pods try to run them at the same time.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing
- Branch is generally up to date with main Github - definitely no conflicts
- No unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- PR description says *what* changed and *why*, with a link to the JIRA story.
- Diff has been checked for unexpected changes being included.
- Commit messages say why the change was made.

### CODE REVIEW

File: lib/tasks/jobs/migrations.rake [hunk 1]
```diff
@@ -1,12 +1,16 @@
 namespace :db do
   namespace :migrations do
```
Comment (MazOneTwoOne) [inline lib/tasks/jobs/migrations.rake:2]: question - why could we not make the namespace more descriptive like `:migrate_again_and_retry` or something along those lines. Maybe it's a bit too wordy.

Comment (patrick-laa) [reply lib/tasks/jobs/migrations.rake:2]: So the "namespace" is a bit generic, like "broadly what is this about". Then line 4, the task name, tries to be a bit descriptive. I thought that "run_with_retry" in the context of "migrations" captured what we wanted here, but yes, you could be as descriptive as you like.
